java/arcs/core/data/proto/HandleProtoDecoder.kt

```diff
@@ -47,11 +47,11 @@ fun List<HandleProto.Capability>.decode() = Capabilities(
  *
  * If a type is not set in the [HandleProto], it is initialized to a newly created TypeVariable.
  */
-fun HandleProto.decode() = Handle(
+fun HandleProto.decode(knownHandles: Map<String, Handle> = mutableMapOf()) = Handle(
     name = name,
     fate = fate.decode(),
     storageKey = storageKey,
     type = if (hasType()) type.decode() else TypeVariable("$name"),
     capabilities = capabilitiesList.decode(),
-    associatedHandles = getAssociatedHandlesList()
+    associatedHandles = associatedHandlesList.map { knownHandles[it]!! }
```

Comment:
Instead of `!!`, consider using `requireNotNull`, and providing a better error message.

Reply:
I didn't know it differed in err messaging, thanks that's good to know.